### ISSUE TITLE
change the adaptors to preserve the "non-dependently typed" property

### ISSUE THREAD

**Issue by ericniebler** [cplusplus/sender-receiver#194]

**Comment by villevoutilainen** [comment]
YES, PLEASE. This allows diagnosing common sender-continuation connections near that connection, not far away where the pipeline gets launched. This is *essential*.

**Comment by lewissbaker** [comment]
I believe that this should be addressed by [P3164R0](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3164r0.html) when that paper is eventually accepted.

See also related issue #206
